File: src/lib/models/DTO/Domain/VideoDto.cs
using System;
using System.Collections.Generic;

namespace DTO.Domain
{
    public class VideoDto : BaseDomainDto
    {
        public string Title { get; set; }
        public string Description { get; set; }
        public short? Part { get; set; }
        public long VidId { get; set; }
        public long ImgId { get; set; }
        public string VideoGroupId { get; set; }

        public UserDto CreationUser{ get; set; }
        public ImgDto Img{ get; set; }
        public VidDto Vid{ get; set; }
        public ICollection<FocusVideoMappingDto> Foci{ get; set; }

        new public bool Validate(bool throwException = false)
        {
            var invalid = base.Validate()
                && string.IsNullOrEmpty(this.Title)
                || string.IsNullOrEmpty(this.Description)
                || (this.Img == null && this.ImgId <= 0)
                || (this.Vid == null && this.VidId <= 0);

            if(invalid && throwException)
            {
                throw new Exception($"Invalid {nameof(VideoDto)} structure");
            }

            return invalid;
        }
    }
}

File: src/ui/asq-ui/src/app/_models/domain/video-dto.ts
import { BaseDomainDto } from './base-domain-dto';
import { UserDto } from './user-dto';
import { ImgDto } from './img-dto';
import { VidDto } from './vid-dto';
import { FocusVideoMappingDto } from './focus-video-mapping-dto';

export class VideoDto extends BaseDomainDto{
    public title: string = '';
    public description: string = '';
    public fileName: string = '';
    public filePath: number = 0;
    public imgId: number = 0;
    public vidId: number = 0;
    public foci: FocusVideoMappingDto[] = null;

    public user: UserDto = null;
    public img: ImgDto = null;
    public vid: VidDto = null;

    constructor(){
        super();
    }
}
